test/functional/rpc_net.py

```diff
@@ -96,5 +100,29 @@ def _test_getpeerinfo(self):
         assert_equal(peer_info[0][0]['addrbind'], peer_info[1][0]['addr'])
         assert_equal(peer_info[1][0]['addrbind'], peer_info[0][0]['addr'])
 
+    def _test_getnodeaddresses(self):
+        self.nodes[0].add_p2p_connection(P2PInterface())
+        network_thread_start()
+        self.nodes[0].p2p.wait_for_verack()
+
+        # send some addresses to the node via the p2p message addr
+        imported_addrs = set(["123.123." + str(i//255) + "." + str(i%255) for i in range(1000)])
```

Comment:
nit: move this into the range loop below and use string formatters rather than concatenation:

```python
        # send some addresses to the node via the p2p message addr
        now = int(time.time())
        msg = msg_addr()
        imported_addrs = []
        for i in range(1000):
            a = "123.123.{}.{}".format(i // 255, i % 256)
            imported_addrs.append(a)
            addr = CAddress()
            ...
```